### PR TITLE
Cancel modal bug

### LINKED ISSUE #202: Cancel Button of Key Modal doesn't work
Hello,

Click on cancel button of Key Modal doesn't close the modal.

I can reproduce the issue by starting the dev server on the master branch. 

Am I missing something?

Seems like a problem with v1.18.1

### PR DESCRIPTION
**Pull request by mac-s-g**
fixes #202

### CODE REVIEW

**Comment by mac-s-g** [comment]
I'll bump patch version as well

File: src/js/index.js [hunk 1]
```diff
@@ -257,7 +257,8 @@ class ReactJsonView extends React.PureComponent {
 
     resetState = () => {
         this.setState({
-            validationFailure: false
+            validationFailure: false,
+            addKeyRequest: false
```
Comment (mac-s-g) [inline src/js/index.js]: this used to work by [setting state directly at render time](https://github.com/mac-s-g/react-json-view/blob/e00a6cf9cc4009dbea0009282dcae6c0623655cd/src/js/index.js#L140) in `index.js`.
it was a hack.  definitely looks better now.

Comment (franklixuefei) [reply src/js/index.js]: awesome!

**Comment by franklixuefei** [comment]
@mac-s-g when you have time, could you publish this?

**Comment by mac-s-g** [comment]
thanks! i'll publish now

**Pull request merged**